### ISSUE TITLE
Console report not working with xUnit 2 and NUnit 3

### ISSUE THREAD

**Issue by mwhelan** [TestStack/TestStack.BDDfy#228]
Using the VS and ReSharper test runners the console report no longer displays with the latest versions of these test frameworks. Any ideas on why that might be?

**Comment by JakeGinnivan** [comment]
Unsure about nUnit, but xunit you ned to log to an injected `ITestOutputHelper`. This is because it supports parallel testing and there is no way to correlate Console logs to the correct test.

Here is how I work around it with Serilog - https://gist.github.com/JakeGinnivan/8fd55a3482b499334d5c4c7e9590d61c

**Comment by robdmoore** [comment]
This seems to work for me with xunit 2:

https://gist.github.com/robdmoore/deeb72772b1c06fbbb130a8b9922e2be

**Comment by robdmoore** [comment]
Some refactoring in Bddfy could make it a lot simpler though. Had to work around the fact all the config is static by using `ThreadLocal`.

**Comment by robdmoore** [comment]
Disclaimer: haven't tested with parallel test runs.

**Comment by robdmoore** [comment]
I've tested it with parallel test runs now - it works (sort of) - for some reason the `Process(story)` method is called multiple times for each test equal to the number of parallel tests being run... Not really sure why

**Comment by robdmoore** [comment]
I've solved it with a bit of a hack for now to keep track of a list of Processed stories. Gist updated to reflect that.

**Comment by robdmoore** [comment]
(see https://gist.github.com/robdmoore/deeb72772b1c06fbbb130a8b9922e2be#file-xunit2bddfytextreporter-cs-L24)

**Comment by jochenparm** [comment]
FYI, console report is actually working just fine on my setup (VS2015 + NUnit3 + NUnit3TestAdapter).

**Comment by robdmoore** [comment]
Is that with parallel test runs?

--

**Comment by jochenparm** [comment]
No, I'm only running sequential tests

**Comment by robdmoore** [comment]
That makes sense then - it's only a problem for parallel tests :)

--

**Comment by jochenparm** [comment]
Oh, ok, I'm sorry. That wasn't clear to me from the discussion.

**Comment by robdmoore** [comment]
It won't work with xunit2 regardless because it no longer picks up Console.WriteLine. I guess Nunit 3 must do it differently :)

--

**Comment by shaynevanasperen** [comment]
I've solved this with a library I wrote: https://github.com/shaynevanasperen/TestStack.BDDfy.Xunit

I've been a bit cheeky by naming the NuGet package with the TestStack.BDDfy prefix. If the maintainers of TestStack.BDDfy object to this, I'm happy to pass ownership to them. Hopefully this library will only be a temporary requirement, until TestStack.BDDfy natively supports Xunit parallel test execution (which might also require changes to Xunit). At least it can possibly serve as a starting point in that direction.
